### PR TITLE
fix newline in raw string in e2e net perf case

### LINKED ISSUE #46083: [e2e test] Networking IPerf failed due to invalid `bandwidth` string with newline
**WHAT**: **Bug report**. 

See the error log below,

```
• Failure [100.294 seconds]
[k8s.io] Networking IPerf [Experimental] [Slow] [Feature:Networking-Performance]
/home/dujun/goproject/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:695
  should transfer ~ 1GB onto the service endpoint 1 servers (maximum of 1 clients) [It]
  /home/dujun/goproject/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/networking_perf.go:154

  May 19 15:48:52.070: Failed parsing value bandwidth port from the string '1176527324
  ' as an integer
```

I assume the output of iperf in my test environment has an extra newline in the value of `bandwidth`, that is 
```
'1176527324
' 
```

What about trim the newline in the returned value before parsing it as an integer? Especially change this line https://github.com/kubernetes/kubernetes/blob/master/test/e2e/util_iperf.go#L100

Like,

```
rawValue = strings.Replace(rawValue, "\n", "", -1)
value, err := strconv.ParseInt(rawValue, 10, 64)
```

### PR DESCRIPTION
**Pull request by m1093782566**
**Which issue this PR fixes** 

fixes #46083

### CODE REVIEW

**Comment by spxtr** [comment]
> fixes #46581

That's this PR!

**Comment by m1093782566** [comment]
@spxtr updated.

**Comment by cblecker** [comment]
@k8s-bot ok to test
/release-note-none

**Comment by m1093782566** [comment]
@cblecker  test flakes?

**Comment by m1093782566** [comment]
@k8s-bot ok to test

**Comment by gyliu513** [comment]
@k8s-bot pull-kubernetes-e2e-kops-aws test this

**Comment by m1093782566** [comment]
ha ha. It seems k8s-bot does response to me. THANKS, @gyliu513

**Comment by gyliu513** [comment]
Yes, for now only Member can run `@k8s-bot ok to test`, but you can run the command that the @k8s-ci-robot provided to re-run the test. ;-)

**Comment by m1093782566** [comment]
@k8s-ci-robot pull-kubernetes-e2e-kops-aws test this

**Comment by m1093782566** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by m1093782566** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by cblecker** [comment]
@m1093782566 The federation test is flaky atm, but also non-blocking. We can leave that failed -- it won't block your PR from merging. Next step is waiting for one of the reviewers to look over your change.

**Comment by m1093782566** [comment]
@cblecker Thanks for reminding.

**Review by cmluciano — DISMISSED** [review]
I'm not sure if this actually would not still generate an error. 

I tried an example in the Go playground [here](https://play.golang.org/p/GsGz0Sg3xG)

**Comment by m1093782566** [comment]
@cmluciano 

Please check https://play.golang.org/p/XuelZdR6a9

**Comment by cmluciano** [comment]
Yep that example works since you convert prior to parsing. I should also ask what platform you are running these tests on and can you provide a test run that fails.

**Comment by m1093782566** [comment]
@cmluciano My environment is centos 7.2 kubeadm environment.

It's a bit difficult to create a failure case in CI environment which I think it's a debain OS. However, I guess trim newline in a string does no harm?

**Comment by cmluciano** [comment]
The tests use a specific library that makes this a little easier. I think just a unit test as you mentioned should suffice.

**Comment by m1093782566** [comment]
@cmluciano Do you mean I should unit test `intOrFail()`?

I just don't know how to write unit test case in e2e framework and how to make e2e run unit test cases? Could you please give me some hints? Thanks very much.

**Comment by cmluciano** [comment]
Sorry, I just realized that this actually WAS the e2e. We could try to stub that int in another case, but it probably is not worthwhile. 

Let's just re-arrange the PR to strip the newline prior to parsing for now and then see what the reviewers think of another test case.

**Comment by m1093782566** [comment]
@sstarcher Please look at here. Did you meet the same issue in AWS/CoreOS setup?

**Comment by m1093782566** [comment]
@cmluciano 

Sorry, but to be honest, I don't very understand your expectation. I can reproduce this issue in my local centos setup, and @sstarcher saw this issue in his  AWS/CoreOS setup. I just don't know much about why it does not occur in k8s CI environment. Should we keep this PR open?

**Comment by m1093782566** [comment]
@bowei 

Would appreciate if you can give your opinions. Thanks a lot :)

**Comment by sstarcher** [comment]
I was using the https://github.com/coreos/tectonic-installer at the time when I reported I saw that issue running the conformance tests.  It was installed in AWS using coreos.

File: test/e2e/network/util_iperf.go [hunk 1]
```diff
@@ -97,6 +97,8 @@ func (s StrSlice) get(i int) string {
 
 // intOrFail is a convenience function for parsing integers.
 func intOrFail(debugName string, rawValue string) int64 {
+	// Trim newline in raw string before parsing it
+	rawValue = strings.Replace(rawValue, "\n", "", -1)
```
Comment (bowei) [inline test/e2e/network/util_iperf.go]: This doesn't seem great to me. We should be Trim()ing the string before handing to this function. Also, this function makes 

"1\n2\n3\n5" become a valid integer.

**Review by bowei — CHANGES_REQUESTED** [review]
I would fix the calling places where the newline is being introduced.

File: test/e2e/network/util_iperf.go (hunk 1 above)
Comment (m1093782566) [reply test/e2e/network/util_iperf.go]: Fixed. PTAL. Thanks!

**Comment by m1093782566** [comment]
Glad to receive your feedback, @bowei 

I fixed the newline in `NewIPerf()`, please take a look when you have a chance. Thanks!

File: test/e2e/network/util_iperf.go [hunk 2]
```diff
@@ -69,6 +69,8 @@ func (i *IPerfResults) ToTSV() string {
 
 // NewIPerf parses an IPerf CSV output line into an IPerfResult.
 func NewIPerf(csvLine string) *IPerfResult {
+	// Trim newline in csv string line before parsing it
```
Comment (bowei) [inline test/e2e/network/util_iperf.go]: This comment is not needed.

Comment (m1093782566) [reply test/e2e/network/util_iperf.go]: Okay, removed this comment.

**Comment by m1093782566** [comment]
@bowei @cmluciano 

All comments are fixed now and this PR has been pending for a while...

**Pull request merged**